### DISCUSSION TITLE
Distributed Arch - Maximum Master capacity/scalability

### DISCUSSION THREAD

**Discussion by EbolaWare** [Security-Onion-Solutions/securityonion#4738] [Unsupported Versions]
Hello All!
I'm looking at implementing SO in a large and complicated network, and it's hard to duplicate [i.e. "not viable"] this network structure/traffic load to stress test before deciding whether or not to swap over. 
I'm hoping that someone out there has done this before, or knows some limits based on code, and can help provide some info. 

1. What are the limits on forwarding nodes or data/traffic flow through the master? Are there limits?
2. If there are limits on the master, is it possible to scale the master node horizontally?

For the purposes of this discussion, lets assume that hardware is "unlimited", and everything that's not a sensor will be bare-metal servers.

EDIT: We are also aware of the scale that ElasticSearch will require, and are ready to accommodate that. **_ES will not be our bottleneck._**

_Thank you in advance for your wisdom!_
**-eW**

**Comment by weslambert** [comment]
There are no limits as to the number of nodes that can reside within a grid -- the main limit you'll find is the amount of traffic you are monitoring relative to the capability of the hardware (before you need to start splitting off into multiple nodes using a flow-based load balancer, etc), or from a search speed or data retention perspective, for example,  if you need to move to multiple search nodes or a hot-warm type of architecture.
